portfolio/target/portfolio-1/calendar.js

```diff
@@ -1,86 +1,141 @@
- * This function assumes that the date is the current date when the function is called. 
+/**
+ * Creates a Date object based on a string that represents a time in HH:MM
+ * format. This function assumes that the date is the current date when the
+ * function is called.
  */
 function getTimeObject(timeString) {
-  const today = new Date(); 
+  const today = new Date();
   const currentYear = today.getFullYear();
   const currentMonth = today.getMonth();
   const currentDate = today.getDate();
-  const timeHour = timeString.split(":")[0];
-  const timeMinute = timeString.split(":")[1];
-  return new Date(
-      currentYear, currentMonth, currentDate, timeHour, timeMinute); 
+  const timeHour = timeString.split(':')[0];
+  const timeMinute = timeString.split(':')[1];
+  return new Date(currentYear, currentMonth, currentDate, timeHour, timeMinute);
 }
 
 /** Creates a single element for a new calendar event. */
 function updateCalendarEventList(newCalendarEvent) {
-  const newEventElement = document.createElement('li');
-  newEventElement.innerHTML = 'New Event: ' + newCalendarEvent.name 
-      + '<br>starts at ' + newCalendarEvent.startTime
-      + '<br>ends at ' + newCalendarEvent.endTime; 
-
+  const newEventCard = document.createElement('div');
+  newEventCard.classList.add('card'); 
+ 
+  const cardBody = document.createElement('div');
+  cardBody.classList.add('card-body');
+  newEventCard.appendChild(cardBody);
+ 
+  const cardTitle = document.createElement('h4');
+  cardTitle.classList.add('card-title');
+  cardTitle.innerText = newCalendarEvent.name;
+  cardBody.appendChild(cardTitle);
+ 
+  const startTimeText = document.createElement('p');
+  startTimeText.classList.add('card-text');
+  startTimeText.innerText = newCalendarEvent.startTime; 
+  cardBody.appendChild(startTimeText); 
+ 
+  const endTimeText = document.createElement('p');
+  endTimeText.classList.add('card-text');
+  endTimeText.innerText = newCalendarEvent.endTime; 
+  cardBody.appendChild(endTimeText); 
+ 
+  const deleteButton = document.createElement('button');
+  deleteButton.classList.add('btn');
+  deleteButton.classList.add('btn-danger');
+  deleteButton.innerText = 'Delete this event'; 
+  cardBody.appendChild(deleteButton); 
+ 
   const eventList = document.getElementById('new-event-list');
   eventList.innterHTML = ''; 
-  eventList.appendChild(newEventElement); 
+  eventList.appendChild(newEventCard); 
+ 
+  deleteButton.onclick = function(newEventCard, cardTitle, startTimeText, endTimeText) {
+    newEventCard.target.closest('div.card').remove(); 
+    console.log(cardTitle); 
+    const event = new CalendarEvent(cardTitle, startTimeText, endTimeText); 
+    allEvents.delete(JSON.stringify(event));
+  }
 }
 
 /** Collects all the events currently displayed on the UI. */
 function collectAllEvents() {
   // A set for all calendar events displayed on the UI.
   // Each element in this set is a Json string. 
   allEventJson = new Set(); 
-
+ 
```

Comment:
Remove extra space.

Reply:
The file is under target, so please ignore.